pkg/client/introduce/client.go

```diff
@@ -52,22 +60,42 @@ func New(ctx Provider) (*Client, error) {
 		Event:   introduceSvc,
 		service: introduceSvc,
 		store:   store,
+		newUUID: func() string { return uuid.New().String() },
```

Comment:
are you expecting a new client to be created for each invocation of introduce?
how does the recipient listen for introduction requests? certainly they cannot creating new clients on every potentially parallel request.

we are building a controller model and we also need to consider other bindings like rest.